File: src/main/java/Schedule.java
import java.util.ArrayList;
import java.util.List;
import java.util.StringJoiner;

public class Schedule {

  private final List<DaySchedule> daySchedules;

  public Schedule() {
    daySchedules = new ArrayList<>();
  }

  public void add(final DaySchedule daySchedule) {
    daySchedules.add(daySchedule);
  }

  public List<DaySchedule> getDaySchedules() {
    return daySchedules;
  }

  @Override
  public String toString() {
    StringJoiner sj = new StringJoiner(", ");
    daySchedules.forEach(daySchedule -> {
      sj.add(daySchedule.toString());
    });
    return sj.toString();
  }
}

File: src/main/java/com/adamkoch/iptables/Schedule.kt
package com.adamkoch.iptables

import com.adamkoch.annotations.Unstable
import java.util.*
import java.util.function.Consumer

/**
 *
 * @since 0.1.0
 * @author aakoch
 */
@Unstable
class Schedule {
    private val dayOfWeekSchedules: MutableList<DayOfWeekSchedule>
    fun add(dayOfWeekSchedule: DayOfWeekSchedule) {
        dayOfWeekSchedules.add(dayOfWeekSchedule)
    }

    fun getDaySchedules(): List<DayOfWeekSchedule> {
        return dayOfWeekSchedules
    }

    override fun toString(): String {
//        return daySchedules.joinToString()
        val sj = StringJoiner(", ")
        dayOfWeekSchedules.forEach(Consumer { dayOfWeekSchedule: DayOfWeekSchedule -> sj.add(dayOfWeekSchedule.toString()) })
        return sj.toString()
    }

    init {
        dayOfWeekSchedules = ArrayList()
    }
}